packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -348,15 +349,29 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 		}
 	}
 
-	protected override reSubmitCore(content: any, localOpMetadata: unknown) {
+	protected override reSubmitCore(content: JsonCompatibleReadOnly, localOpMetadata: unknown) {
+		assert(isJsonObject(content), "expected content to be json object");
```

Comment:
isJsonObject actually tells you if something which is already known to be json compatible, is an object, so its better to say:
```suggestion
		assert(isJsonObject(content), "expected content to an object");
```

Since json is a text format (so technically all json data is strings), and you are partly using this check to ensure what you have is not a string, this clarification is extra useful.

Reply:
fixed assert message